plugins/woocommerce-admin/client/layout/controller.js

```diff
@@ -481,14 +540,21 @@ window.wpNavMenuClassChange = function ( page, url ) {
 		element.classList.add( 'menu-top' );
 	} );
 
-	const pageUrl =
-		url === '/'
-			? 'admin.php?page=wc-admin'
-			: 'admin.php?page=wc-admin&path=' + encodeURIComponent( url );
-	const currentItemsSelector =
-		url === '/'
-			? `li > a[href$="${ pageUrl }"], li > a[href*="${ pageUrl }?"]`
-			: `li > a[href*="${ pageUrl }"]`;
+	let pageUrl = 'admin.php?page=wc-admin',
+		currentItemsSelector = `li > a[href$="${ pageUrl }"], li > a[href*="${ pageUrl }?"]`;
+
+	if ( '/' !== url ) {
+		pageUrl = 'admin.php?page=wc-admin&path=' + encodeURIComponent( url );
+		const splitUrl = url.split('/');
+		if ( '' === splitUrl[0] ) {
+			splitUrl.splice(0, 1);
+		}
+		currentItemsSelector =
+			splitUrl.length > 1
+				? `li > a[href*="${ encodeURIComponent( splitUrl[0] ) }"]`
+				: `li > a[href*="${ pageUrl }"]`;
+	}
```

Comment:
The intention behind this change is to highlight the parent menu item when we're viewing one of its children.

`url` is the `path` parameter passed in the WC Admin URL. If `url` contains a slash, we split it and use the first non-empty fragment as the selector for the current menu item.

So if we're viewing `wp-admin/admin.php?page=wc-admin&path=%2Fmarketplace%2Fextensions`, we'll highlight the menu item for its parent page, `wp-admin/admin.php?page=wc-admin&path=%2Fmarketplace`.

Reply:
Removed this change: using tabs in a single page component is less problematic.